packages/lesswrong/components/seasonal/solsticeSeason/GlobePopup.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState, useMemo } from "react";
+import { useStyles } from "@/components/hooks/useStyles";
+import GroupLinks from "@/components/localGroups/GroupLinks";
+import { StyledMapPopupContent } from "@/components/localGroups/StyledMapPopup";
+import { postGetPageUrl } from "@/lib/collections/posts/helpers";
+import { fixedPositionEventPopupStyles } from "../HomepageMap/HomepageCommunityMap";
+import { useFloating, autoUpdate, offset, flip, shift } from '@floating-ui/react-dom';
+import type { VirtualElement } from '@floating-ui/dom';
+
+
+export const GlobePopup = ({document, screenCoords, onClose}: {
+  document: PostsList;
+  screenCoords: { x: number; y: number };
+  onClose: () => void;
+}) => {
+  const fixedPopupClasses = useStyles(fixedPositionEventPopupStyles);
+  const [windowWidth, setWindowWidth] = useState(typeof window !== 'undefined' ? window.innerWidth : 0);
```

Comment:
this is an alarming way to initialize state and i have no idea if it's safe or not